server/src/test/java/com/google/coffeehouse/servlets/CreateClubServletTest.java

```diff
@@ -0,0 +1,174 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.coffeehouse.servlets;
+
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.coffeehouse.util.IdentifierGenerator;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringReader;
+import java.io.StringWriter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+
+/**
+ * Unit tests for {@link CreateClubServlet}.
+ */
+public class CreateClubServletTest {
+  private static final String NAME = "Club Name";
+  private static final String DESCRIPTION = "Club Description";
+  private static final String IDENTIFICATION_STRING = "predetermined-identification-string";
+  private static final String BOOK_TITLE = "Book Name";
+  private static final String BOOK_AUTHOR = "Book Author";
+  private static final String BOOK_ISBN = "978-3-16-148410-0";
+  private static final String MINIMUM_JSON = String.join("\n", 
+      "{",
+      "  \"name\" : \"" + NAME + "\",",
+      "  \"currentBook\" : {",
+      "    \"title\" : \"" + BOOK_TITLE + "\"",
+      "  }",
+      "}");
+  private static final String MAXIMUM_JSON = String.join("\n", 
+      "{",
+      "  \"name\" : \"" + NAME + "\",",
+      "  \"description\" : \"" + DESCRIPTION + "\",",
+      "  \"contentWarnings\" : [\"1\", \"2\"],",
+      "  \"currentBook\" : {",
+      "    \"title\" : \"" + BOOK_TITLE + "\",",
+      "    \"author\" : \"" + BOOK_AUTHOR + "\",",
+      "    \"isbn\" : \"" + BOOK_ISBN + "\"",
+      "  }",
+      "}");
+  private static final String NO_NAME_JSON = String.join("\n", 
+      "{",
+      "  \"currentBook\" : {",
+      "    \"title\" : \"" + BOOK_TITLE + "\"",
+      "  }",
+      "}");
+  private static final String NO_BOOK_JSON = String.join("\n", 
+      "{",
+      "  \"name\" : \"" + NAME + "\"",
+      "}");
+  private static final String SYNTACTICALLY_INCORRECT_JSON = "{\"name\"";
+
+  private CreateClubServlet CreateClubServlet;
+  private final LocalServiceTestHelper helper = new LocalServiceTestHelper();
+  private StringWriter stringWriter = new StringWriter();
+
+  @Mock private HttpServletRequest request;
+  @Mock private HttpServletResponse response;
+
+  @Before
+  public void beforeTest() throws IOException {
+    helper.setUp();
+
+    IdentifierGenerator idGen = Mockito.mock(IdentifierGenerator.class);
+    Mockito.when(idGen.generateId()).thenReturn(IDENTIFICATION_STRING);
```

Comment:
nit: It's often better to import static functions individually, so `import static org.mockito.Mockito.when;` and then you can just call it as `when()` instead of `Mockito.when()`

Reply:
Thanks for the tip! The necessary tests have been updated.